src/preamble.js

```diff
@@ -2401,7 +2403,13 @@ function integrateWasmJS(Module) {
   Module['asmPreload'] = Module['asm'];
 
   // Memory growth integration code
+
+  var defaultReallocBuffer = Module['reallocBuffer'];
+
   Module['reallocBuffer'] = function(size) {
+    if (finalMethod === 'asmjs') {
```

Comment:
It's not global, it's inside `integrateWasmJS` scope. Or do you mean that is too global?